src/main/scala/io/citrine/lolo/bags/Bagger.scala

```diff
@@ -280,6 +281,32 @@ class BaggedModel[+T: ClassTag](
 
     res.asInstanceOf[BaggedResult[T]]
   }
+
+  /**
+    * Compute Shapley feature attributions for a given input
+    *
+    * @param input for which to compute feature attributions.
+    * @param omitFeatures feature indices to omit in computing Shapley values
+    * @return array of Shapley feature attributions, one per input feature, each a vector of
+    *         One DenseVector[Double] per feature, each of length equal to the output dimension.
+    *         The output dimension is 1 for single-task regression, or equal to the number of classification categories.
+    */
+  override def shapley(input: Vector[Any], omitFeatures: Set[Int] = Set()): Option[Vector[DenseVector[Double]]] = {
+    val ensembleShapley = models.map(model => model.shapley(input, omitFeatures))
+    if (!ensembleShapley.head.isDefined) {
+      None
+    }
+    assert(ensembleShapley.forall(x=>x.isDefined))
+    def sumReducer(a: Option[Vector[DenseVector[Double]]],
+                   b: Option[Vector[DenseVector[Double]]]): Option[Vector[DenseVector[Double]]] = {
+      (a ++ b).reduceOption[Vector[DenseVector[Double]]]{case (x,y) => x.zip(y).map{case (v1,v2) => (v1 + v2)}}
```

Comment:
This seems like it could be written more simply

Reply:
Yeah, I'm not thrilled with it. But the option makes things a little annoying, as does  the vector of vectors. If model.shapley returned an `Option[DenseMatrix[Double]]`, then `Seq[DenseMatrix[Double]]` would appear at first glance to be nearly compatible with `breeze.stats.mean`, except that breeze doesn't support tensors :(
Do you have another suggestion?

Reply:
Yes, but after trying them out in IntelliJ they don't seem to work. I was assuming that `DenseVector` behaved the same as `Vector`.